forms/gridfield/GridFieldDetailForm.php

```diff
@@ -385,20 +385,22 @@ public function ItemEditForm() {
 				$actions->push(new LiteralField('cancelbutton', $text));
 			}
 		}
+		
+		// If we are creating a new record in a has-many list, then
+		// pre-populate the record's foreign key.
+		if($list instanceof HasManyList && !$this->record->isInDB()) {
+			$key = $list->getForeignKey();
+			$id = $list->getForeignID();
+			$this->record->$key = $id;
```

Comment:
Does it work if you retain the `if(!$this->record->isInDB()) {` check? I can forsee change detection being an issue here.

Besides, you only need this for new records (not in db) right?